modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateRequest.java

```diff
@@ -119,6 +120,9 @@ public boolean isSimulate() {
     }
 
     public void setSimulate(boolean simulate) {
+        if (simulate) {
```

Comment:
This only applies when the setter is called, what do you think about applying this during the constructor if and only if request is null.

I suggest updating the validation logic to complain if the request isn't SIMULATED request, otherwise it might not be obvious to callers about the request is being ignored.

Reply:
`setSimulate` must be called if the caller is setting `simulate` to true. None of the constructors allow setting this value as part of the constructor. 

I agree that it would be better as part of the constructor to ensure it can never be null even if this class is changed in the future.